src/UsersModule/Components/LandingPage/LandingPage.tsx

```diff
@@ -1,17 +1,941 @@
-import React from 'react'
-import { useNavigate } from "react-router-dom"
+import {
+  Box,
+  Button,
+  Container,
+  Grid,
+  TextField,
+  Typography,
+} from "@mui/material";
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { baseUrl } from "../../../Constants/Components/Urls";
+import { FavoriteOutlined, Star } from "@mui/icons-material";
+import Visibility from "@mui/icons-material/Visibility";
+import styles from "./LandingPage.module.css";
+import hotel1 from "../../../assets/hotel1.png";
+import hotel2 from "../../../assets/hotel2.png";
+import hotel3 from "../../../assets/hotel3.png";
+import hotel4 from "../../../assets/hotel4.png";
+
+import house1 from "../../../assets/house1.png";
+import house2 from "../../../assets/house2.png";
+import house3 from "../../../assets/house3.png";
+import house4 from "../../../assets/house4.png";
+
+import landingImg from "../../../assets/landingImg.png";
+import { useForm } from "react-hook-form";
+import happyFamily from "../../../assets/happyFamily.png";
+import AnonymousUserAlert from "../AnonymousUserAlert/AnonymousUserAlert";
+import { toast } from "react-toastify";
+import NoData from "../../../SharedModule/Components/NoData/NoData";
+import Loading from "../../../SharedModule/Components/Loading/Loading";
 
 export default function LandingPage() {
-  const navigate=useNavigate();
+  const navigate = useNavigate();
+  const [ads, setAds] = React.useState([]);
+  let fiveAds = ads?.slice(0, 5);
+  const [isLoading, setIsLoading] = useState(false);
+
+
+  const [anonymousAlert, setAnonymousAlert] = useState(false);
+  const handleCloseAnonymousAlert = () => setAnonymousAlert(false);
```

Comment:
We can also make both functions  into one like this:

```jsx
  const handleAnonymousAlert = (state:"close"|"open") =>{
    return ()=>{
      setAnonymousAlert("close" === state ? false: true )
    }
  }
```
and then use them like this:
```jsx
  onClick={handleAnonymousAlert("close")}
  onClick={handleAnonymousAlert("open")}
```

This technique is called currying. it is useful in some situations, but both approaches are fine.